### DISCUSSION TITLE
Does fastapi-users work with pymongo or it explicitly requires beanie?

### DISCUSSION THREAD

**Discussion by matteius** [fastapi-users/fastapi-users#1181] [Q&A] [answered]
Howdy, I have been building out a Fast API for sensor data collection and I use pymongo and custom db projections to sample the data for the time window selected when viewing historical data.   I am interested in using fastapi-users but less interested in adding the beanie dependency, mostly because I don't know it/if I need it and want to continue using pymongo without having to rework my existing projections.   Can anyone provide some advice or background on this topic?

**Comment by frankie567** [comment, marked as answer]
We only support Beanie as backend for MongoDB. That said, you have two solutions:

* Implement a database adapter for FastAPI Users with pure Mongo. It's quite straightforward, it mainly involves to implement the following class: https://github.com/fastapi-users/fastapi-users/blob/master/fastapi_users/db/base.py
* Use Beanie only for FastAPI Users and keep pure Mongo for your own logic. Don't see any blockers with this approach.
